File: cloud/Server/Contracts/Permissions/CreatePermissionOptions.cs
// Copyright © Spatial Corporation. All rights reserved.

namespace Spatial.Cloud.Contracts.Permissions;

/// <summary>
/// Configurable options for a permission.
/// </summary>
public class CreatePermissionOptions
{
    /// <summary>
    /// The role the permission is granted to.
    /// </summary>
    public string Role { get; set; }

    /// <summary>
    /// The scope to grant access to.
    /// </summary>
    public string Scope { get; set; }
}

File: interface/data/permissions/CreatePermissionOptions.ts
// Copyright © Spatial Corporation. All rights reserved.

/**
 * Configurable options for a new permission.
 */
export type CreatePermissionOptions = {
  /**
   * The role the permission is granted to.
   */
  role: string;

  /**
   * The scope to grant access to.
   */
  scope: string;
};
